models/models_methods.go

```diff
@@ -1166,34 +1166,49 @@ func (sIDData StepIDData) IsUniqueResourceID() bool {
 // ----------------------------
 // --- BuildRunResults
 
-// IsStepLibUpdated ...
 func (buildRes BuildRunResultsModel) IsStepLibUpdated(stepLib string) bool {
 	return (buildRes.StepmanUpdates[stepLib] > 0)
 }
 
-// IsBuildFailed ...
 func (buildRes BuildRunResultsModel) IsBuildFailed() bool {
 	return len(buildRes.FailedSteps) > 0
 }
 
-// HasFailedSkippableSteps ...
 func (buildRes BuildRunResultsModel) HasFailedSkippableSteps() bool {
 	return len(buildRes.FailedSkippableSteps) > 0
 }
 
-// ResultsCount ...
 func (buildRes BuildRunResultsModel) ResultsCount() int {
 	return len(buildRes.SuccessSteps) + len(buildRes.FailedSteps) + len(buildRes.FailedSkippableSteps) + len(buildRes.SkippedSteps)
 }
 
+func (buildRes BuildRunResultsModel) IsBuildAbortedWithTimeout() bool {
+	for _, stepResult := range buildRes.FailedSteps {
+		if stepResult.Status == StepRunStatusAbortedTimeout {
+			return true
```

Comment:
Is this true though? What happens if a step that is marked as `is_skippable` is aborted this way? In this case the workflow will still be successful, no?

Reply:
Failed but skipped steps are collected separately: https://github.com/bitrise-io/bitrise/blob/9127b0e96e9392257ff85e57b8dd267fd2ca4506/models/models.go#L141